flutter-app/lib/camera.dart

```diff
@@ -83,54 +93,74 @@ class _CameraState extends State<Camera> {
 
   @override
   Widget build(BuildContext context) {
+    if (widget.isUploading && widget.resultList != null) {
+      widget.resultList
+          .forEach((eachImage) => {uploadImageToStorage(eachImage.path)});
+    }
     return Scaffold(
       key: _scaffoldKey,
       appBar: AppBar(
         backgroundColor: Theme.of(context).accentColor,
         title: Text('Capture sample for ${widget.label}'),
       ),
-      body: Container(
-        decoration: new BoxDecoration(color: Colors.black),
-        child: Column(
-          children: <Widget>[
-            Expanded(
-              child: Container(
-                child: Padding(
-                  padding: const EdgeInsets.all(1.0),
-                  child: Center(
-                    child: _cameraPreviewWidget(),
-                  ),
-                ),
-                decoration: BoxDecoration(
-                  color: Colors.black,
-                  border: Border.all(
-                    color:
-                        controller != null && controller.value.isRecordingVideo
-                            ? Colors.redAccent
-                            : Colors.black45,
-                    width: 2.0,
-                  ),
-                ),
+      body: widget.isUploading
+          ? Center(
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                crossAxisAlignment: CrossAxisAlignment.center,
+                children: <Widget>[
+                  Text('Uploading Images in the background'),
+                  MaterialButton(
+                    onPressed: () {
+                      Navigator.pop(context);
+                    },
+                    child: Text('OK'),
+                  )
```

Comment:
Can this be a snackbar or a dialog instead? IMO, we can just navigate to the previous screen & show a snackbar informing user that images are being uploaded in the background.